src/plugins/vis_type_table/public/table_vis_response_handler.ts

```diff
@@ -39,60 +40,62 @@ export interface TableGroup {
 }
 
 export interface Table {
-  $parent?: TableGroup;
   columns: Input['columns'];
   rows: Input['rows'];
 }
 
-export function tableVisResponseHandler(table: Input, dimensions: any): TableContext {
-  const converted: TableContext = {
-    tables: [],
-  };
+export function tableVisResponseHandler(input: Input, dimensions: Dimensions): TableContext {
+  let table: Table | undefined;
+  let tables: TableGroup[] = [];
+  let direction: TableContext['direction'];
 
   const split = dimensions.splitColumn || dimensions.splitRow;
 
   if (split) {
-    converted.direction = dimensions.splitRow ? 'row' : 'column';
+    tables = [];
+    direction = dimensions.splitRow ? 'row' : 'column';
     const splitColumnIndex = split[0].accessor;
     const splitColumnFormatter = getFormatService().deserialize(split[0].format);
-    const splitColumn = table.columns[splitColumnIndex];
+    const splitColumn = input.columns[splitColumnIndex];
     const splitMap = {};
     let splitIndex = 0;
 
-    table.rows.forEach((row, rowIndex) => {
+    input.rows.forEach((row, rowIndex) => {
       const splitValue: any = row[splitColumn.id];
```

Comment:
Can we type this better? As it's used in `hasOwnProperty` a little down below, we can probably cast it to a string anyway.

Reply:
enhanced types in the file -> https://github.com/elastic/kibana/pull/70801/commits/ada2cae7793e0bbae324d0e8865fe5deadbaa7be